moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_calcs.h

```diff
@@ -89,7 +89,12 @@ class JogCalcs
 
   void publishWarning(bool active) const;
 
-  bool checkIfJointsWithinURDFBounds(trajectory_msgs::JointTrajectory_<std::allocator<void>>& new_joint_traj);
+  // Checks the current robot state for compliance with joint limits.
+  // If out of bounds, returns a halt trajectory that will stop the robot
+  // @param halting_joint_traj - the trajectory to execute to stop the robot if joints are out of bounds
```

Comment:
I think the functionality here is muddled because there's also a suddenHalt() function. It's not clear to me why some trajectories are halted within checkIfJointsWithinURDFBoundsAndCreateHaltTrajectory() but others use suddenHalt(). Also, there's issue #1844.

None of those are caused by this PR but I need to sort them out.